tests/shared/src/test/scala/scala/meta/tests/io/NIOPathTest.scala

```diff
@@ -79,13 +81,25 @@ class NIOPathTest extends FunSuite {
   test(".relativize(Path)") {
     assert(abs.relativize(abs.resolve("qux")) == Paths.get("qux"))
   }
-  test(".toUri") {
+  test("file.toUri") {
     assert(file.toUri.getPath.endsWith("build.sbt"))
     // NOTE: Paths API seems to work inconsistently under Scala Native.
     // [info] - .toUri *** FAILED ***
     // [info]   "/Users/eburmako/Projects/scalameta/project" did not end with "project/" (NIOPathTest.scala:84)
     // assert(project.toUri.getPath.endsWith("project/"))
   }
+
+  test("jar.toUri") {
+    if (scala.meta.internal.platform.isJVM) {
+      val jar = Files.createTempDirectory("scalameta").resolve("foo.jar")
+      FileIO.withJarFileSystem(AbsolutePath(jar), true, true) { root =>
+        val hello = root.resolve("hello")
+        Files.createDirectories(hello.toNIO)
+        assert(hello.toURI.toString.endsWith("hello"))
```

Comment:
I'm not sure if the expected behavior here is `endsWith("hello/")` or `endsWith("hello")`. I don't mind either way however, as long as `toURI` doesn't crash.